### PR TITLE
Simplification

### PR DESCRIPTION
**Pull request by th3o6a1d**
Hi team,

I took a birds eye view of the SDK and noticed opportunities to make the function names and behavior more internally consistent. There are essentially two types of responses from the SDK, "Orders" and "Items." They both can be searched, and so their search filters should look similar. The mint function needs to take a bare minimum number of inputs and handle the minting transactions/signing etc on the backend.

Let me know what you think

Jason

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,59 @@
...
+### Mint an NFT
+
+```
+const rarible = new RaribleSDK();
+const myNFT = {
+    supply: 5,
+    metadata: {
+        name: "Inception",
+        description: "The beginning.",
+        image: "...png"
```
Comment (benefacto) [inline README.md:25]: Will this be a file path? If so, I think we should have a more specific example like: `/some/path/image.png` which also begs the question as to whether we want to support globbing.

File: src/models/mint.ts [hunk 2]
```diff
@@ -1,85 +1,37 @@
...
-  id: string;
-  contract: string;
-  tokenId: string;
-  creator?: string;
-  supply: number;
-  lazySupply: number;
-  owners: string[];
-  royalties: PartOwner[];
-  pending: ItemTransfer[];
+export interface Signature {
```
Comment (benefacto) [inline src/models/mint.ts:33]: Do we need an object that only has a single property? Or can it just be a `string` named `signature`?

**Review by benefacto — APPROVED** [review]
I left some feedback but this looks really good me! I especially like the improved readme.

File: src/models/orders.ts [hunk 3]
```diff
@@ -39,45 +33,75 @@ export interface OrderDataLegacy {
...
+  type: "bid_by_maker";
+}
+
+export type OrderFilter =
+  | Sell
+  | SellByMaker
+  | SellByItem
+  | SellByItem
+  | BidByItem
+  | BidByMaker;
```
Comment (carlosfebres) [inline src/models/orders.ts:101]: I don't thin interface for each type isn't required here 🤔

**Review by carlosfebres — CHANGES_REQUESTED** [review]
I like this simplification! but let's sync before merging this

File: README.md [hunk 4]
```diff
@@ -1,42 +1,59 @@
...
 
 > Disclaimer: This is still a Work in progress, and no logic has yet been developed for the SDK. This interface will serve as a cornerstone to integrate with the functions relevant to end users / developers of the Rarible ecosystem. 
 
-## Overview
 
-With `todo-name:RaribleSDK.js`, you can mint, buy, sell, and bid on NFTs on Ethereum (ERC-721 and ERC-1155) by accessing the most versatile functions of the Rarible Protocol. This allows developers like you to quickly iterate and build solutions for their specific use cases.
+## Installation
 
-This interface aims to cover the entire [Protocol Flow](https://docs.rarible.com/#protocol-flow). 
+`npm i --save <name>`
```
Comment (carlosfebres) [inline README.md:12]: This project doesn't need to be installed

**Review by benefacto — COMMENTED** [review]
I think that @carlosfebres incorporated most of your changes in b98cd57aaa7227c5c154a7c5ec3aaf44c1c56108 so this might be closed out.

**Pull request closed without merging**